R/datasets.R

```diff
@@ -0,0 +1,627 @@
+#' a Datastore and the datastore type must be of a SQL kind. Column data types are
+#' read from data types in SQL query result. Providing `set_column_types` will
+#' override the data type  for the specified columns in the returned Tabular Dataset.
+#'
+#' @param query A SQL-kind datastore and a query
+#' @param validate Boolean to validate if data can be loaded from the returned dataset.
+#' Defaults to True. Validation requires that the data source is accessible from
+#' the current compute.
+#' @param set_column_types A named list to set column data type, where key is
+#' column name and value is data type.
+#' @return The Tabular Dataset object
+#' @export
+#' @md
+create_tabular_dataset_from_sql_query <- function(query, validate = TRUE,
+                                                  set_column_types = NULL) {
+  azureml$core$dataset$Dataset$Tabular$from_sql_query(query, validate,
+                                                      set_column_types)
+}
+
+#' Drop the specified columns from the dataset.
+#'
+#' @description
+#' Drop the specified columns from the dataset. If a timeseries column is dropped,
+#' the corresponding capabilities will be dropped for the returned dataset as well.
+#'
+#' @param dataset The Tabular Dataset object.
+#' @param columns A name or a list of names for the columns to drop.
+#' @return A new TabularDataset object with the specified columns dropped.
+#' @export
+#' @md
+drop_columns_from_dataset <- function(dataset, columns) {
+  dataset$drop_columns(columns)
+}
+
+#' Keep the specified columns and drops all others from the dataset.
+#'
+#' @description
+#' Keep the specified columns and drops all others from the dataset.
+#' If a timeseries column is dropped, the corresponding capabilities will be
+#' dropped for the returned dataset as well.
+#'
+#' @param dataset The Tabular Dataset object
+#' @param columns The name or a list of names for the columns to keep.
+#' @param validate Indicates whether to validate if data can be loaded from the
+#' returned dataset. The default is False. Validation requires that the data
+#' source is accessible from current compute.
+#' @return A new Tabular Dataset object with only the specified columns kept.
+#' @export
+#' @md
+keep_columns_from_dataset <- function(dataset, columns, validate = FALSE) {
+  dataset$keep_columns(columns, validate)
+}
+
+#' Filter Tabular Dataset with time stamp columns after a specified start time.
+#'
+#' @description
+#' Filter Tabular Dataset with time stamp columns after a specified start time.
+#'
+#' @param dataset The Tabular Dataset object
+#' @param start_time The lower bound for filtering data.
+#' @param include_boundary Boolean indicating if the row associated with the
+#' boundary time (``start_time``) should be included.
+#' @return The filtered Tabular Dataset
+#' @export
+#' @md
+filter_dataset_after_time <- function(dataset, start_time,
```

Comment:
can you confirm if R times get converted to a Python datetime or NumPy datetime64 by reticulate? The Python methods that have `start_time`/`end_time` parameters expect a Python datetime object

Reply:
Yep - it does since reticulate 1.11. https://cran.r-project.org/web/packages/reticulate/news/news.html